lib/h3core.js

```diff
@@ -843,6 +895,91 @@ function h3Distance(origin, destination) {
+ * origin index.
+ * - Failure may occur if the index is too far away from the origin
+ * or if the index is on the other side of a pentagon.
+ * - This function is experimental, and its output is not guaranteed
+ * to be compatible across different versions of H3.
+ * @static
+ * @param  {H3Index} origin      Origin H3 index
+ * @param  {H3Index} destination H3 index for which to find relative coordinates
+ * @return {Object}              Coordinates as an `{i, j}` pair
+ * @throws {Error}               If the IJ coordinates cannot be calculated
+ */
+function experimentalH3ToLocalIj(origin, destination) {
+    const ij = C._malloc(SZ_COORDIJ);
+    const [oLower, oUpper] = h3IndexToSplitLong(origin);
+    const [dLower, dUpper] = h3IndexToSplitLong(destination);
+    const retVal = H3.experimentalH3ToLocalIj(oLower, oUpper, dLower, dUpper, ij);
+    const coords = readCoordIJ(ij);
+    C._free(ij);
+    // Return the pair, or throw if an error code was returned.
+    // Switch statement and error codes cribbed from h3-java's implementation.
+    switch (retVal) {
+        case 0:
+            return coords;
+        case 1:
+            throw new Error('Incompatible origin and index.');
+        default:
+        case 2:
+            throw new Error(
+                'Local IJ coordinates undefined for this origin and index pair. ' +
+                    'The index may be too far from the origin.'
+            );
+        case 3:
+        case 4:
+        case 5:
+            throw new Error('Encountered possible pentagon distortion');
+    }
+}
+
+/**
+ * Produces an H3 index for IJ coordinates anchored by an origin.
+ *
+ * - The coordinate space used by this function may have deleted
+ * regions or warping due to pentagonal distortion.
+ * - Coordinates are only comparable if they come from the same
+ * origin index.
+ * - Failure may occur if the index is too far away from the origin
+ * or if the index is on the other side of a pentagon.
+ * - This function is experimental, and its output is not guaranteed
+ * to be compatible across different versions of H3.
+ * @static
+ * @param  {H3Index} origin     Origin H3 index
+ * @param  {Object} coords      Coordinates as an `{i, j}` pair
+ * @return {H3Index}            H3 index at the relative coordinates
+ * @throws {Error}              If the H3 index cannot be calculated
+ */
+function experimentalLocalIjToH3(origin, coords) {
+    // Validate input coords
+    if (!coords || typeof coords.i !== 'number' || typeof coords.j !== 'number') {
+        throw new Error('Coordinates must be provided as an {i, j} object');
+    }
+    // Allocate memory for the CoordIJ struct and an H3 index to hold the return value
+    const ij = C._malloc(SZ_COORDIJ);
+    const out = C._malloc(SZ_H3INDEX);
+    storeCoordIJ(ij, coords);
+    const [lower, upper] = h3IndexToSplitLong(origin);
+    const retVal = H3.experimentalLocalIjToH3(lower, upper, ij, out);
```

Comment:
Same nit, not super important.

Reply:
Ok, updated this for my new functions, will do the rest in a cleanup diff.